### PR TITLE
Feature/#9 adding an issue

### PR DESCRIPTION
**Pull request by reissjohnson**
connects #9 

- Still a bit of a WIP but its at the point where we could ship it for a working add issue feature.

### CODE REVIEW

**Review by reissjohnson — COMMENTED** [review]
Change ordering of adding a new ticket, so they do not get sent to the bottom.

File: app/controllers/api/board_tickets_controller.rb [hunk 1]
```diff
@@ -1,5 +1,4 @@
-class Api::BoardTicketsController < Api::BaseController
-
+class Api::BoardTicketsController < AuthenticatedController
```
Comment (rjpaskin) [inline app/controllers/api/board_tickets_controller.rb:1]: API controllers are generally for `flight_plan_cli` and use tokens to authenticate - recommend adding the `create` action to the existing `BoardTicketsController` instead?

File: app/services/ticket_creation_service.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        remote_state: remote_ticket[:state]
+      )
+      board.board_tickets.create!(ticket: ticket, swimlane: board.swimlanes.first)
+    end
+  end
+
+  private
+
+  def board_repo
+    BoardRepo.find(@repo_id)
```
Comment (rjpaskin) [inline app/services/ticket_creation_service.rb]: Nit: use the `attr_reader` here?

File: app/services/ticket_creation_service.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  def create_ticket!
+    Ticket.transaction do
+      ticket = repo.tickets.new(remote_title: title, remote_body: description)
+      ticket.save!
+      remote_ticket = create_remote_ticket
+      ticket.update!(
+        remote_id: remote_ticket[:id],
+        remote_number: remote_ticket[:number],
+        remote_state: remote_ticket[:state]
+      )
```
Comment (rjpaskin) [inline app/services/ticket_creation_service.rb]: What do you think about creating the remote ticket first, then using `Ticket.import` to get it into the database? Saves having to update the ticket - and having it in the database in an incomplete state.

File: app/javascript/action_creators/index.js [hunk 4]
```diff
@@ -75,13 +81,27 @@ export const ticketDragged = ({ source, destination, draggableId }) => dispatch
...
+                    destination: source,
+                    boardTicketId
+                })
             )
         );
 };
 
+export const ticketCreated = ticketAttributes => dispatch => {
+    return createTicket(ticketAttributes).then(response =>
+        dispatch(ticketCreation(response))
```
Comment (rjpaskin) [inline app/javascript/action_creators/index.js]: Handle error responses and promise rejections?

**Review by rjpaskin — CHANGES_REQUESTED** [review]
Think we need to do something with the prettier config - I think my editor might be using the settings from MyRewards, so maybe we should add those settings and rebase this branch so that we don't get all the changes/conflicts?

**Review by rjpaskin — APPROVED** [review]
Subject to reformatting to match existing style (and subsequent PR to get whole codebase formatted the same)

**Comment by rjpaskin** [comment]
Closed in favour of #142

**Pull request closed without merging**